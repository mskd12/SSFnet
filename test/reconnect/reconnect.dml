# 
# reconnect.dml
#
# Three ASes connected in a line, each with one router.
#  
#
#       router      router       router
# 
#        1:1 }------{ 2:1 }------{ 3:1
#

_schema [ _find .schemas.Net ]

Net [  # the all-encompassing Net

  frequency 1000000000   # nanosecond simulation frequency

  bgpoptions [
    validation_test 15      # prints messages associated with this test
  ]

  # ASes 2 and 3 are practically identical, but AS 1 has a keep-alive interval
  # which is much longer than the hold interval, which will cause the
  # connection to time out from the point of view of the router in AS 2.
  Net [
    id 1
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [
            as 2  address 1(1)  use_return_address 1(1)
            hold_time 90  keep_alive_time 200  mrai 30
            infilter  [ _extends .filters.permit_all ]
            outfilter [ _extends .filters.permit_all ]
          ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  Net [
    id 2
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 1  address 1(1)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
          neighbor [ as 3  address 1(1)  use_return_address 1(2)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ idrange [ from 1 to 2 ] ]
    ]
  ]

  Net [
    id 3
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 2  address 1(2)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  # connect the ASes together
  link [ attach 1:1(1) attach 2:1(1) delay 0.01 ]
  link [ attach 2:1(2) attach 3:1(1) delay 0.01 ]

] # end of all-encompassing Net
