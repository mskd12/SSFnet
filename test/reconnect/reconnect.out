~#
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~# RECONNECT Validation Test: successful if three steps printed below
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~#
~# 1. AS Y: ended session with AS X
~# 2. AS Y: reestablished session with AS X
~# 3. AS Z: received second advertisement of route to AS X
~#             *** TEST SUCCESSFUL ***
~#
